javascript/007-high-score-board/high-score-board.js

```diff
@@ -0,0 +1,77 @@
+/// <reference path="./global.d.ts" />
+// @ts-check
+
+/**
+ * Creates a new score board with an initial entry.
+ *
+ * @returns {Record<string, number>} new score board
+ */
+export function createScoreBoard() {
+  return { 'The Best Ever': 1000000 }
+}
+
+/**
+ * Adds a player to a score board.
+ *
+ * @param {Record<string, number>} scoreBoard
+ * @param {string} player
+ * @param {number} score
+ * @returns {Record<string, number>} updated score board
+ */
+export function addPlayer(scoreBoard, player, score) {
+  scoreBoard[player] = score;
+  return scoreBoard;
+}
+
+/**
+ * Removes a player from a score board.
+ *
+ * @param {Record<string, number>} scoreBoard
+ * @param {string} player
+ * @returns {Record<string, number>} updated score board
+ */
+export function removePlayer(scoreBoard, player) {
+  delete scoreBoard[player];
+  return scoreBoard;
+}
+
+/**
+ * Increases a player's score by the given amount.
+ *
+ * @param {Record<string, number>} scoreBoard
+ * @param {string} player
+ * @param {number} points
+ * @returns {Record<string, number>} updated score board
+ */
+export function updateScore(scoreBoard, player, points) {
+  scoreBoard[player] = scoreBoard[player] + points;
+  return scoreBoard;
+}
+
+/**
+ * Applies 100 bonus points to all players on the board.
+ *
+ * @param {Record<string, number>} scoreBoard
+ * @returns {Record<string, number>} updated score board
+ */
+export function applyMondayBonus(scoreBoard) {
+  const mondayBonus = 100;
+
+  for (let player in scoreBoard) {
+    if (Object.prototype.hasOwnProperty.call(scoreBoard, player)) {
```

Comment:
Explain please why you need this condition

Reply:
It's EsLint rule look below. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
https://eslint.org/docs/latest/rules/no-prototype-builtins